src/camera.py

```diff
@@ -0,0 +1,247 @@
+        else:
+            self._pan_offset = int(np.clip(degree + self._pan_offset,
+                                                    self.MIN_PAN_CALI_DEGREE,
+                                                    self.MAX_PAN_CALI_DEGREE))
+
+        self._db.set('pan_offset', self._pan_offset)
+
+
+    def get_pan_angle(self):
+        """
+        Gets the current pan angle of the camera.
+
+        :return: The current pan angle (in degrees).
+        :rtype: int
+        """
+        return self._current_pan_angle
+
+
+    def tilt(self, degree):
+        """
+        Tilts the camera.
+
+        Controls the vertical movement of the camera in [0, 150] degrees.
+        0 is directly in front of the car, 90 faces upwards, and 150 faces the
+        Raspberry Pi. The degrees will be clamped to [0, 150].
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        if self._camera_type == ABSOLUTE:
+            self._current_tilt_angle = np.clip(degree, self.MIN_DEGREE,
+                                                       self.MAX_TILT_DEGREE)
+        else:
+            self._current_tilt_angle = np.clip(degree + self._current_tilt_angle,
+                                                        self.MIN_DEGREE,
+                                                        self.MAX_TILT_DEGREE)
+
+        self._tilt_servo.write(np.clip(self._current_tilt_angle +
+                                       self._tilt_offset,
+                                       self.MIN_DEGREE,
+                                       self.MAX_TILT_DEGREE))
+
+
+    def calibrate_tilt(self, degree):
+        """
+        Calibrates the camera's tilting servo.
+
+        Writes a new tilt offset value to the config file and changes the current
+        tilt offset. The calibration value can be negative.
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        if self._camera_type == ABSOLUTE:
+            self._tilt_offset = int(np.clip(degree, self.MIN_TILT_CALI_DEGREE,
+                                                    self.MAX_TILT_CALI_DEGREE))
+        else:
+            self._tilt_offset = int(np.clip(degree + self._tilt_offset,
+                                                     self.MIN_TILT_CALI_DEGREE,
+                                                     self.MAX_TILT_CALI_DEGREE))
+
+        self._db.set('tilt_offset', self._tilt_offset)
+
+
+    def get_tilt_angle(self):
+        """
+        Gets the current tilt angle of the camera.
+
+        :return: The current tilt angle (in degrees).
+        :rtype: int
+        """
+        return self._current_tilt_angle
+
+
+def main():
+    """
+    Demonstrates the camera's panning and tilting functionality.
+
+    Creates a camera object and pans from 0 to 180, then from 180 to 0. The
+    camera then tilts from 0 to 150, then 150 to 0.
+    """
+    camera_smoothing = 0.05
```

Comment:
Should use a better name. This isn't being used to "smooth," just adds a delay in the for loops.